### PR TITLE
1858 bash cleanup

### PR DESCRIPTION
**Pull request by jonahx**
### What does this PR do?
- _What's changed? Why were these changes made?_

  - Fixes all shellcheck errors in `ci/test`.
  - Cleans up code generally
  - Adds comments
  - Fixes a handful of bugs

- _How should the reviewer approach this PR, especially if manual tests are required?_

This is a just a straight read the old code and new code and compare on one long file.

### What ticket does this PR close?

#1858 (I will update the issue title to reflect current reality)

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: ci/test [hunk 1]
```diff
@@ -7,9 +7,17 @@
 # this script.  We should fix these errors and turn them on.
 set -e
 
+# TODO: Documentation of higher level dependencies.  Partial list:
+# 1. All the Azure env args assumed to exist.
+# 2. Same for OIDC / GCP.
+# 3. Why are we calling via sh in Jenkinsfile when these are bash files?
```
Comment (micahlee) [inline ci/test:13]: It looks like it does use the shebang to determine the interpreter: https://www.jenkins.io/doc/pipeline/steps/workflow-durable-task-step/#sh-shell-script

**Review by micahlee — APPROVED** [review]
This all looks reasonable. I'm not super familiar with the finer details of bash. Is there anything particularly different or risky to review more closely?

Did you test each of the test paths to confirm they work as expected?

**Comment by jonahx** [comment]
@micahlee Thanks for the quick review.  I will add a summary of the major bash practices that went into this when I write the squashed commit message tomorrow.

> Did you test each of the test paths to confirm they work as expected?

What were you thinking besides the pipeline continuing to work?

**Comment by micahlee** [comment]
@jonahx 
> What were you thinking besides the pipeline continuing to work?

I wasn't thinking of anything in particular. It's not obvious to me what is just syntax change versus which changes introduce a behavior change (if any). 

If you're good with what you see in the pipeline, I don't have any objection. 🙂

**Review by micahlee — APPROVED** [review]
LGTM!

**Pull request merged**